### PR TITLE
cephadm: install podman from the Kubic project

### PR DESCRIPTION
**Pull request by mgfritch**
ppa:projectatomic is no longer maintained, updates are now provided via the Kubic project
    
Fixes: https://tracker.ceph.com/issues/48072
Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
https://github.com/ceph/ceph/pull/33367#discussion_r380695261

**Comment by bk201** [comment]
- CI branch: [wip-kiefer-testing-2020-11-04-1131](https://github.com/ceph/ceph-ci/commits/wip-kiefer-testing-2020-11-04-1131)
- [CI packages](https://shaman.ceph.com/builds/ceph/wip-kiefer-testing-2020-11-04-1131/)
- QA runs: **FAIL**
  Suite|Total|Pass|Fail|Dead
  -|-|-|-|-
  [rados:cephadm](https://pulpito.ceph.com/kchang-2020-11-04_06:38:14-rados:cephadm-wip-kiefer-testing-2020-11-04-1131-distro-basic-smithi)|56|52|3|1

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-11-06_15:51:09-rados-wip-kefu-testing-2020-11-06-1936-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-11-08_14:53:34-rados-wip-kefu-testing-2020-11-07-2116-distro-basic-smithi/

failure tracked by

- https://tracker.ceph.com/issues/46318

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -4643,6 +4643,50 @@ class Apt(Packager):
...
+        keyring = self.kubric_repo_gpgkey_path()
+        call_throws(['apt-key', '--keyring', keyring, 'add', tmp_key.name])
+
+        content = 'deb %s /\n' % self.kubic_repo_url()
+
+        logger.info('Installing repo file at %s...' % self.kubic_repo_path())
+        with open(self.kubic_repo_path(), 'w') as f:
+            f.write(content)
+
+    def rm_kubic_repo(self):
```
Comment (tchaikov) [inline src/cephadm/cephadm:4679]: shall we call this method after installing podman?

**Review by bk201 — APPROVED** [review]

File: src/cephadm/cephadm (hunk 1 above)
Comment (mgfritch) [reply src/cephadm/cephadm:4679]: we might want the repo later during `apt upgrade`, but maybe we should call this during the `rm-repo` command?

```
ubuntu@ubuntu:~$ sudo ~/cephadm rm-repo 
Removing repo GPG key /etc/apt/trusted.gpg.d/kubic.release.gpg...
Removing repo at /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list...
```

**Comment by mgfritch** [comment]
jenkins test api

**Comment by tchaikov** [comment]
jenkins test api

**Review by tchaikov — APPROVED** [review]

**Comment by mgfritch** [comment]
https://pulpito.ceph.com/mgfritch-2020-11-10_01:16:20-rados:cephadm-wip-mgfritch-testing-2020-11-09-1644-distro-basic-smithi/

Failure:
https://tracker.ceph.com/issues/48171

**Pull request merged**